src/providers/ContentShareProvider/state.tsx

```diff
@@ -90,38 +198,102 @@ export function reducer(
       };
     }
     case ContentActionType.UPDATE: {
-      const { isLocalUser, tileState } = payload as UpdatePayload;
-      const { tileId } = state;
+      const { isLocalUser, tileState } = payload;
+      const tileIdToUpdate = tileState.tileId!;
+      const attendeeIdToUpdate = tileState.boundAttendeeId!;
 
-      if (
-        tileId === tileState.tileId ||
-        (tileId && tileId > tileState.tileId!)
-      ) {
+      // Skip updates to existing content share tiles
+      // This ensures we only process new content shares and ignore updates to existing ones
+      // (such as video quality changes, pauses/resumes at the video tile level, etc.)
+      if (state.tileIdToAttendeeId[tileIdToUpdate.toString()]) {
         return state;
       }
 
+      const collections = addTileToCollections(
+        state,
+        tileIdToUpdate,
+        attendeeIdToUpdate
+      );
+
+      const backwardCompatibility = updateBackwardCompatibility(
```

Comment:
may have a better var and method name. maybe `updatedState` and `updateForBackwardCompatibility`?

Reply:
This is to handle backward compatibility for existing `tileId` and `sharingAttendeeId`.

```ts
const backwardCompatibility = 
  {
    tileId: number | null;
    sharingAttendeeId: string | null;
  }
```

Reply:
I think const `updatedStateForBackwardCompatibility` and `updateForBackwardCompatibility()` best describe the purpose, but they are a little verbose.

Reply:
updated to a better naming in my opinion